### DISCUSSION TITLE
HttpFoundation HTTP/1.1 messaging protocols

### DISCUSSION THREAD

**Discussion by DrechslerDerek** [symfony/symfony#54210] [Q&A]
PSR-7 follows HTTP/1.1 messaging protocols outlined in RFC 7230 and RFC 7231, which are from 2014. The protocols outlined these documents are obsoleted by RFC 9110 and RFC 9112. Will PSR-7 be updated to be more aligned with these protocols? I'm not sure, but there seems to be no recent news?

My question is, does Symfony HttpFoundation, the Response and Request objects, follow RFC 9110 and RFC 9112, or RFC 7230 and RFC 7231 and comply with PSR-7 (it would appear not because the PSR-7 component is still available?). What HTTP/1.1 messaging protocols does HttpFoundation follow, is it entirely dependent on SAPI?

Thanks!

Follow up: What specifically makes HttpFoundation non psr-7 compliant and what are the differences, advantages, disadvantages. They both follow HTTP/1.1 standards and the RFCs specify changes/updates to HTTP/1.1, 2 and 3 standards.

**Comment by stof** [comment]
The main difference between PSR-7 and HttpFoundation is that PSR-7 defines immutable objects for Request and Response while the HttpFoundation objects are mutable.

**Comment by stof** [comment]
Regarding the messaging protocols, the new RFCs have not introduced substantial changes compared to the old ones. The new RFCs are mostly about separating the HTTP semantics (shared between HTTP 1, 1.1, 2 and 3) from the HTTP/1.1 syntax. The actual changes are mostly about clarifying the spec for some edge cases. Those changes did not require changes in HttpFoundation
